torch/_inductor/codegen/triton.py

```diff
@@ -1259,7 +1260,14 @@ def final_argreduce(buffer, result_var, value, index):
                 self.compute.writeline(
                     f"{accumulator} = tl.where({cond}, {updated}, {accumulator})"
                 )
-                self.suffix.writeline(f"{result_var} = {final_reduction(accumulator)}")
+
+                if src_dtype == torch.bool:
+                    # tl.reduce doesn't support tl.int1
+                    accumulator = f"{accumulator}.to(tl.int8)"
+                    result_type = triton_compute_type(dtype)
+                    self.suffix.writeline(f"{result_var} = {final_reduction(accumulator)}.to({result_type})")
```

Comment:
This is only really used for `aten.any`. It changes the final reduction of a non-persistent reduction from
```python
tmp5 = triton_helpers.max(_tmp5, 1)[:, None]
```

```python
tmp5 = triton_helpers.max(_tmp5.to(tl.int8), 1)[:, None].to(tl.int1)
```

Reply:
This is good to put as a comment in the code!